### DISCUSSION TITLE
SQL variables

### DISCUSSION THREAD

**Discussion by lkordos** [duckdb/duckdb#9935] [Q&A]
I want to declare a local SQL variable but I am getting the following error:
"syntax error near declare"

**I am using the following syntax:**
declare @myCount int;

I cannot find anything about SQL variables in the duckdb documentation. 

What is the correct syntax to use?

Thanks.

**Comment by Mause** [comment]
DuckDB does not support user defined variables, hence the lack of docs

**Reply by aheilbut** [reply]
Do you know if this is a feature that is deliberately not supported by design, or is it just something that hasn't (yet) been implemented?

**Reply by aheilbut** [reply]
oh!  I see that it is actually already in the works.. https://github.com/duckdb/duckdb/pull/13084

**Comment by lkordos** [comment]
ok, thank you.

As a suggestion: if it doesn't exist already, having a list of differences against SQL standard would be helpful.
